force_wfmanager/ui/setup/mco/mco_parameter_view.py

```diff
@@ -0,0 +1,312 @@
+                        editor=InstanceEditor(),
+                        style="custom",
+                        show_label=False
+                    ),
+                    springy=True,
+                ),
+                HGroup(
+                    Item('add_parameter_button',
+                         springy=True,
+                         enabled_when='object.parameter_entity_creator'
+                                      '.model is not None'),
+                    Item('remove_parameter_button',
+                         enabled_when='selected_parameter is not None'),
+                    show_labels=False,
+                ),
+            )
+        )
+
+        return traits_view
+
+    #: Defaults
+    def _parameter_entity_creator_default(self):
+        """Returns an entity creator containing parameter types
+        from all installed plugins"""
+
+        if self.model is not None:
+            visible_factories = [
+                f for f in self.model.factory.parameter_factories
+                if f.ui_visible
+            ]
+
+            parameter_entity_creator = NewEntityCreator(
+                factories=visible_factories,
+                dclick_function=self._dclick_add_parameter,
+                factory_name='MCO Parameter',
+                config_visible=False
+            )
+            return parameter_entity_creator
+
+    def _parameter_model_views_default(self):
+        """Creates a list of MCOParameterModelViews for each
+        model.parameter"""
+        parameter_model_views = []
+
+        if self.model is not None:
+            # Add all MCO parameters as ModelViews
+            parameter_model_views += [
+                MCOParameterModelView(model=parameter)
+                for parameter in self.model.parameters
+            ]
+
+        return parameter_model_views
+
+    def _selected_parameter_default(self):
+        """Default value for selected_kpi"""
+        if len(self.parameter_model_views) > 0:
+            return self.parameter_model_views[0]
+
+    #: Listeners
+    @on_trait_change('model')
+    def update_parameter_entity_creator(self):
+        """ Update the entity creator based on the MCO factory """
+        self.parameter_entity_creator = (
+                self._parameter_entity_creator_default()
+        )
+
+    # Update parameter_model_views when model changes
+    @on_trait_change('model.parameters')
+    def update_parameter_model_views(self):
+        """ Update the MCOParameterModelView(s) """
+        self.parameter_model_views = self._parameter_model_views_default()
+
+        # Update the selected_parameter view
+        if len(self.parameter_model_views) == 0:
+            self.selected_parameter = None
+
+    # Workflow Validation
+    @on_trait_change('parameter_model_views.verify_workflow_event')
+    def received_verify_request(self):
+        """Pass on request for verify_workflow_event"""
+        self.verify_workflow_event = True
+
+    #: Button actions
```

Comment:
I think a comment with `#:` gets picked up by the traits specific sphinx extension we use. Though just ignore this if that was intentional!

Reply:
Ah, good to know - there might be a lot of cleaning up to do in which case

Reply:
I'd take a look at the documentation output, it might not be getting picked up. Wouldn't worry about fixing it if it isn't.

I know it definitely does for attributes e.g.
```
#: Here's what this trait does
important_trait = Str()
```
but maybe functions aren't included.

Reply:
Ok, great - I'll make a new PR with a load of cleaning for sphinx comments anyway